src/config.ts

```diff
@@ -89,31 +85,89 @@ export class Config {
 
     return cfg;
   }
-}
 
-/// Gas limits for various transactions.
-const gasLimits = {
-  createOrg: 1_200_000,
-};
+
+}
 
 function isMetamaskInstalled(): boolean {
   const { ethereum } = window;
   return Boolean(ethereum && ethereum.isMetaMask);
 }
 
+function isWalletConnectConnected(): boolean {
+  const newWalletConnect = (): WalletConnect => {
+    return new WalletConnect({
+      bridge: "https://radicle.bridge.walletconnect.org",
+    });
+  };
+  walletConnect = newWalletConnect();
+
+  if (walletConnect.connected) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
 export async function getConfig(): Promise<Config> {
-  let config: Config;
 
-  if (isMetamaskInstalled()) {
-    const metamask = new ethers.providers.Web3Provider(window.ethereum);
-    const network = await metamask.ready;
-    config = new Config(network, metamask, metamask.getSigner());
+  let network = { name: "homestead", chainId: 1 };
+
+  const metamask = isMetamaskInstalled() ? new ethers.providers.Web3Provider(window.ethereum) : null;
+
+  if (metamask) {
+    network = await metamask.ready;
+  }
+
+
+  const networkConfig = (<Record<string, any>> config)[network.name];
+  if (!networkConfig) {
+    throw `Network ${network.name} is not supported`;
+  }
+
+  function getProvider(): ethers.providers.JsonRpcProvider {
+    // Use Alchemy in production, on mainnet.
+    let provider;
+
+    if (network.name === "homestead" && import.meta.env.PROD) {
+      provider = new ethers.providers.AlchemyProvider(network.name, config.alchemy.key);
+    } else if (isMetamaskInstalled()) {
+      provider = new ethers.providers.Web3Provider(window.ethereum);
+    }
+    if (! provider) {
+      throw `No Web3 provider available.`;
+    }
```

Comment:
This function can be cleaned up by removing `let provider` and just having an `else` clause which throws.

Reply:
Unresolved :warning:

Reply:
Still unresolved!